src/provider/BaseBibleAPIProvider.test.ts

```diff
@@ -0,0 +1,145 @@
+import { BibleAPIDotComProvider } from './BibleAPIDotComProvider'
+import { IBibleVersion } from '../interfaces/IBibleVersion'
+
+describe('BaseBibleAPIProvider - Ref.ly URL validation', () => {
+  // Create a mock bible version for testing
+  const mockBibleVersion: IBibleVersion = {
+    key: 'kjv',
+    name: 'King James Version',
+    code: 'en',
+    apiSource: {
+      name: 'Bible API',
+      apiUrl: 'https://bible-api.com',
+    },
+  }
+
+  // All books from the reflyAbbreviations array
+  const bibleBooks = [
```

Comment:
This may be a duplication, because the JSON files from different trasnlations have contain the book names